theseus/optimizer/nonlinear/nonlinear_optimizer.py

```diff
@@ -220,6 +220,9 @@ def _optimize_impl(
     def compute_delta(self, **kwargs) -> torch.Tensor:
         pass
 
+    def compute_samples(self, n_samples: int, T: float) -> torch.Tensor:
```

Comment:
Leave a comment here as discussed to say what's the intended use, in case it's confusing for a user to see a sampling function inside an optimizer.